src/sentry/integrations/slack/views/link_identity.py

```diff
@@ -40,47 +48,102 @@ class SlackLinkIdentityView(BaseView):
     Django view for linking user to slack account. Creates an entry on Identity table.
     """
 
+    _METRICS_SUCCESS_KEY = "sentry.integrations.slack.link_identity_view.success"
+    _METRICS_FAILURE_KEY = "sentry.integrations.slack.link_identity_view.failure"
+
     @method_decorator(never_cache)
-    def handle(self, request: Request, signed_params: str) -> HttpResponse:
+    def dispatch(self, request: HttpRequest, *args, **kwargs) -> HttpResponseBase:
         try:
+            signed_params = kwargs.pop("signed_params")
             params = unsign(signed_params)
-        except (SignatureExpired, BadSignature):
+        except (SignatureExpired, BadSignature) as e:
+            _logger.warning("dispatch.signature_error", exc_info=e)
+            metrics.incr(self._METRICS_FAILURE_KEY, tags={"error": str(e)}, sample_rate=1.0)
             return render_to_response(
                 "sentry/integrations/slack/expired-link.html",
                 request=request,
             )
+        except KeyError as e:
+            _logger.warning("dispatch.key_error", exc_info=e)
+            metrics.incr(self._METRICS_FAILURE_KEY, tags={"error": str(e)}, sample_rate=1.0)
+            return render_error_page(
+                request,
+                status=400,
+                body_text="HTTP 400: Missing required 'signed_params' parameter",
+            )
+
+        try:
+            organization, integration, idp = get_identity_or_404(
+                ExternalProviders.SLACK,
+                request.user,
+                integration_id=params["integration_id"],
+            )
+        except Http404:
+            _logger.exception(
+                "get_identity_error", extra={"integration_id": params["integration_id"]}
+            )
+            metrics.incr(self._METRICS_FAILURE_KEY + ".get_identity", sample_rate=1.0)
+            raise
+
+        _logger.info("get_identity_success", extra={"integration_id": params["integration_id"]})
+        metrics.incr(self._METRICS_SUCCESS_KEY + ".get_identity", sample_rate=1.0)
+        params.update({"organization": organization, "integration": integration, "idp": idp})
+        return super().dispatch(request, params=params)
```

Comment:
do we need to update params or could we just pass them into super().dispatch() as specific key word arguments? `super().dispatch(org=organization)`